mustache.lisp

```diff
@@ -546,7 +546,28 @@ The syntax grammar is:
 (defun read-partial (filename context)
   (declare (type (or string pathname) filename)
            (type context context))
-  (let ((from-context (context-get filename (partials context))))
+  
+  (let* ((var-name
+           (cond
+             ;; This block implements optional Dynamic Names
+             ;; from this specification:
+             ;; https://github.com/mustache/spec/blob/master/specs/~dynamic-names.yml
+             ((and (typep filename 'string)
+                   (not (zerop (length filename)))
+                   (char= (elt filename 0)
+                          #\*))
+              (let* ((stripped-filename (subseq filename 1))
+                     (dynamic-name (context-get stripped-filename
+                                                context)))
+                (unless dynamic-name
+                  (error "Unable to find dynamic partial name ~S in a context."
+                         stripped-filename))
```

Comment:
The spec says

> Failed resolution of the dynamic name SHOULD result in nothing being rendered.

The tests also include

> The empty string should be used when the named partial is not found.

Do you want to keep the error behavior? I'm fine to merge as-is since this is allowed by the spec.

Reply:
I think we better to keep to the behaviour specified in spec.

However, personally I don't like it because eating errors silently is not the best default behaviour. But thanks to the god we are using Common Lisp, which allows us to use such powerful tools as signals. I propose to signal a non-error signal in case if partial name was not found. This way, template user might wrap it with a `handler-case` and choose to make an error out of the signal if he wants a behaviour other than default.

What do you think?

Reply:
By the way, it will be cool to implement the similar trick signaling about missing variables.

Also we can provide restarts to `use-value` instead of missing variable.

Reply:
I think both are very good ideas.